tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -628,6 +731,45 @@ policy_refresh_cagg_add_internal(Oid cagg_oid, Oid start_offset_type, NullableDa
 	JsonbValue *result = pushJsonbValue(&parse_state, WJB_END_OBJECT, NULL);
 	Jsonb *config = JsonbValueToJsonb(result);
 
+	PolicyRefreshOffsetOverlapResult res = policy_refresh_cagg_check_for_overlaps(cagg, config, 0);
+	switch (res)
+	{
+		case POLICY_REFRESH_OFFSET_OVERLAP:
+			break;
+		case POLICY_REFRESH_OFFSET_OVERLAP_EQUAL:
+			if (if_not_exists)
+			{
+				ereport(NOTICE,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("an identical continuous aggregate policy already exists for "
+								"\"%s\", "
+								"skipping",
+								get_rel_name(cagg->relid)),
+						 errdetail("A policy with the same start and end offset already exists for "
+								   "continuous aggregate \"%s\"",
+								   get_rel_name(cagg->relid))));
+				PG_RETURN_INT32(-1);
+			}
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+
+					 errmsg("continuous aggregate refresh policy already exists for "
+							"\"%s\", ",
+							get_rel_name(cagg->relid)),
```

Comment:
Suggest you use the same wording for both the notice and the error, with the exception of the "skipping" last. Also, a typo.
```suggestion
					 errmsg("an identical continuous aggregate refresh policy already exists for "
							"\"%s\"",
							get_rel_name(cagg->relid)),
```